src/main/java/com/example/springboot/dao/BookDAOService.java

```
@@ -0,0 +1,188 @@
+	private static final String QUERY_DELETE = "DELETE FROM `books` WHERE id=?;";
+	
+	private static final String ID = "id";
+	private static final String TITLE = "title";
+	private static final String AUTHOR = "author";
+	private static final String READER_ID = "reader_id";
+	
+	/**
+	 * This method create and return a connection
+	 * @return Connection
+	 * @throws SQLException 
+	 */
+	private Connection getConnection() throws SQLException {
+		return DriverManager.getConnection(URL, USERNAME, PASSWORD);
+	}
+	
+	/**
+	 * This method create new instance of Book and return it
+	 * @return Book
+	 * @throws SQLException 
+	 */
+	private Book getBook(ResultSet resultSet) throws SQLException {
+		return new Book(resultSet.getInt(ID), resultSet.getString(TITLE), resultSet.getString(AUTHOR), resultSet.getInt(READER_ID));
+	}
+	
+	/**
+	 * This method return all books from database
+	 * @return List<Book>
+	 */
+	@Override
+	public List<Book> findAll() {
+		List<Book> resultList = new ArrayList<>();
+		
+		try (Connection connection = getConnection()) {
+			PreparedStatement statement = connection.prepareStatement(QUERY_FIND_ALL);
+			ResultSet resultSet = statement.executeQuery();
+			
+			while(resultSet.next()) {
+				resultList.add(getBook(resultSet));
+			}
+			
+			return resultList;
+			
+		} catch(SQLException | RuntimeException e) {
+			LOGGER.error(e.getMessage());
+		}
+		
+		return null;
+	}
+	
+	/**
+	 * This method looking for a book in database by given id
+	 * @param id
+	 * @return Book
+	 */
+	@Override
+	public Book selectById(Integer id) {
+		
+		try (Connection connection = getConnection()) {
+			PreparedStatement statement = connection.prepareStatement(QUERY_SELECT_BY_ID);
+			statement.setInt(1, id);
+			ResultSet resultSet = statement.executeQuery();
+			
+			resultSet.next();
+			Book book = getBook(resultSet);
+			
+			return book;
+			
+		} catch (SQLException | RuntimeException e) {
```

yep, it is the best practice to swallow all kind of exceptions and don't share it with other world. `My services works without error, as I AM A PROGRAMMER`